### PR TITLE
fix(ocrvs-5086): show different label when father address matches

### PR DESCRIPTION
**Pull request by makelicious**
Show matching label depending on father's primary address selection.
- Extend `conditionalCase` type, allow passing `Conditional[]`

preview-father-details

I could not get to a nicely working solution without utilising the `Conditional` type. I opted for not trying to modify the `primaryAddress` `previewGroup` since it was much more involved.

Let me know what you think. ~~I left a todo comment which I intend to fix if you find the approach ok.~~

### CODE REVIEW

**Review by Zangetsu101 — APPROVED** [review]
The changes look good! It's better this way than to handling the form label & review label separately. One other thing, although it's not mentioned in the ticket, in death form we also have a similar case for informant & spouse if I'm not wrong. They should have a field like "place of residence same as deceased". Let's add the conditionals for them as well please!

File: src/utils/address-utils.ts [hunk 1]
```diff
@@ -1092,34 +1092,37 @@ function getAddress(
...
 }
+
+/**
+ * Adds provided conditionals to each field. Mutates the given array.
+ */
+function addConditionalsToFields(
+  fields: SerializedFormField[],
+  conditionalCase: string | Conditional[]
+) {
+  fields.forEach((field) => {
```
Comment (rikukissa) [inline src/utils/address-utils.ts:1116]: Was there some other mutation happening on a higher level that forced this function also to mutate the inner array values?

Comment (makelicious) [reply src/utils/address-utils.ts:1116]: I will check! I opted for caution if there was some performance reason for mutating.

Comment (rikukissa) [reply src/utils/address-utils.ts:1116]: If you see code with side effects in our codebase you can safely assume it's either:
1. Something higher in the call stack is built using mutation and the author didn't have time to fix the root issue
2. Side-effectful code got past code review unnoticed

We advocate for immutable values and anything else should be a rare exception

Comment (makelicious) [reply src/utils/address-utils.ts:1116]: Clear! On this particular case, We are mutating on higher levels. Taking into account the 1.5. delivery, I suggest I'll merge this one after the new changes have been reviewed and check whether I could get rid of the mutations.

**Comment by makelicious** [comment]
Added same conditionals for informant and spouse fields on 'Death' event
preview-father-dead

**Pull request merged**